lib/vrl/compiler/src/expression/object.rs

```diff
@@ -1,25 +1,26 @@
 use std::{collections::BTreeMap, fmt, ops::Deref};
 
 use crate::{
-    expression::{Expr, Resolved},
+    expression::{Expr, Literal, Resolved},
     state::{ExternalEnv, LocalEnv},
+    value::VrlValueConvert,
     vm::OpCode,
     Context, Expression, TypeDef, Value,
 };
 
 #[derive(Debug, Clone, PartialEq)]
 pub struct Object {
-    inner: BTreeMap<String, Expr>,
+    inner: Vec<(Expr, Expr)>,
```

Comment:
This is something we should have discussed in the RFC.

I believe that we should not (yet, at least) support template strings in object literals.

That is, this is allowed:

```coffee
foo = "foo {{bar}}"
```

This is not:

```coffee
foo = { "foo {{bar}}": true }
```

There might come a time when we do want to allow that, but I don't think now is the time. It would probably happen if/when we allow `."foo {{bar}}" = true` as well.
```

Reply:
I'm on the fence about this. To me the object key there is just a regular old string and so should support the interpolation. I should be able to use any expression that returns a string there. I see that currently doesn't work though (e.g. `foo = { upcase("bar"): 123 }` is invalid) though so I can also see the case to disallow it for consistency.

Reply:
Yeah, agreed 👍

Reply:
Have reverted all these changes. Object keys are now parsed differently in `parser.lalrpop` so they are essentially treated as escaped raw strings.